Removed deprecated getdefaultlocale (#2345)

teodorstelian opened the pull request:
## Description

Tautulli.py used locale.getdefaultlocale() in line 73
```python
 plexpy.SYS_LANGUAGE, plexpy.SYS_ENCODING = locale.getdefaultlocale()
``` 
This method is deprecated.

```
DeprecationWarning: 'locale.getdefaultlocale' is deprecated and slated for removal in Python 3.15. Use setlocale(), getencoding() and getlocale() instead. plexpy.SYS_LANGUAGE, plexpy.SYS_ENCODING = locale.getdefaultlocale()
``` 

I changed the line with:
```python
 plexpy.SYS_LANGUAGE, plexpy.SYS_ENCODING = locale.getlocale()
 if not plexpy.SYS_ENCODING:
     plexpy.SYS_ENCODING = locale.getencoding()
``` 

### Issues Fixed or Closed

- Fixes #2345 

## Type of Change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code

JonnyWong16 commented:
`locale.getlocale()` does not return RFC-1766 language codes on Windows.

```python
>>> import locale
>>> locale.getdefaultlocale()
<stdin>:1: DeprecationWarning: 'locale.getdefaultlocale' is deprecated and slated for removal in Python 3.15. Use setlocale(), getencoding() and getlocale() instead.
('en_CA', 'cp1252')
>>> locale.getlocale() 
('English_Canada', '1252')
>>> locale.getencoding()
'cp1252'
```

https://github.com/python/cpython/issues/82986

teodorstelian commented:
So, after a lot of digging into this problem on Windows, I added a special case for WIndows, I don't know how optimal this case is but I added a If for the windows os

```python
if sys.platform == 'win32':
 # Get the kernel of the OS and the default UI Language to locate the language code in Windows
 windll = ctypes.windll.kernel32
 language_code = locale.windows_locale[windll.GetUserDefaultUILanguage()]
 encoding = "cp1252"
```

I tested on my Windows 11 machine and I got:

![image_2024-07-12_014606299](https://github.com/user-attachments/assets/0f047f57-0dbf-48c5-b2fa-4e3874d50000)

JonnyWong16 commented on Tautulli.py:
```diff
@@ -69,8 +70,30 @@ def main():
+
+        # If encoding is not returned, get the encoding
+        if not encoding:
+            encoding = locale.getencoding()
```
`locale.getencoding()` is not available on Python <3.11.

JonnyWong16 commented:
> So, after a lot of digging into this problem on Windows, I added a special case for WIndows, I don't know how optimal this case is but I added a If for the windows os
> 
> ```python
> if sys.platform == 'win32':
>  # Get the kernel of the OS and the default UI Language to locate the language code in Windows
>  windll = ctypes.windll.kernel32
>  language_code = locale.windows_locale[windll.GetUserDefaultUILanguage()]
>  encoding = "cp1252"
> ```
> 
> I tested on my Windows 11 machine and I got:
> 
> ![image_2024-07-12_014606299](https://private-user-images.githubusercontent.com/51364577/348018076-0f047f57-0dbf-48c5-b2fa-4e3874d50000.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3OTk5MTQsIm5iZiI6MTcyMDc5OTYxNCwicGF0aCI6Ii81MTM2NDU3Ny8zNDgwMTgwNzYtMGYwNDdmNTctMGRiZi00OGM1LWIyZmEtNGUzODc0ZDUwMDAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEyVDE1NTMzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5ZjZkYWQ5YmIzMjJmMTY4ZmY3ZWEyMGJkYWUxNDg2MGFkMTdjZjczYWI2YjZlODcyZDA1N2E4MDgyMzdmZGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qSj7HsofQ0wJjbNMCpiwgKSYZJCfoFWC4QPzUCpI0YA)

Not quite right. `en_CA` vs `en_GB`

```python
>>> locale.getdefaultlocale()
<stdin>:1: DeprecationWarning: 'locale.getdefaultlocale' is deprecated and slated for removal in Python 3.15. Use setlocale(), getencoding() and getlocale() instead.
('en_CA', 'cp1252')

>>> import ctypes
>>> windll = ctypes.windll.kernel32
>>> lang_id = windll.GetUserDefaultUILanguage()
>>> locale.windows_locale.get(lang_id, '')
'en_GB'
```

teodorstelian commented:
On my side, I have my windows languages on English (UK) and I found another method inside the windll.kerner32. Its one called GetUserDefaultLCID(), this should return the default locale for the current user. I also noticed GetThreadUILanguage and  GetSystemPreferredUILanguages.

GetThreadUILanguage - returns the Windows display language so maybe that can also be used instead of GetUserDefaultLCID().

For now, I haven't found anything else

JonnyWong16 commented on Tautulli.py:
```diff
@@ -69,8 +70,30 @@ def main():
+            language_code = locale.windows_locale.get(lang_id, '')
+
+            # Set encoding to Windows default encoding
+            encoding = "cp1252"
```
Hardcoded value or `locale.getpreferredencoding()`?

teodorstelian commented:
You are correct to remove the hardcoded part. I made the changes.
